src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx

```diff
@@ -118,28 +114,40 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
     [accountId, assetNamespace, assetReference, chainId, defaultAccountId, highestBalanceAccountId],
   )
 
-  const opportunityData = useAppSelector(state =>
+  const earnOpportunityData = useAppSelector(state =>
     selectEarnUserStakingOpportunityByUserStakingId(state, opportunityDataFilter),
   )
 
+  const opportunitiesMetadata = useAppSelector(state => selectStakingOpportunitiesById(state))
+
+  const opportunityMetadata = useMemo(
```

Comment:
assignment better than cast here?

Reply:
Unfortunately we can't do this, this guy is sad and not narrowed down with an assignment:

<img width="1342" alt="image" src="https://user-images.githubusercontent.com/17035424/212164683-5397f8aa-914c-41ef-adaf-6b20a16cf510.png">

Reply:
actually no this narrowing is required, smort man @gomesalexandre